src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -124,6 +125,11 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
                                     err.Format()));
         requestorCore->mProviderNodeId = nodeId;
 
+        MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
+        CopySpanToMutableSpan(response.updateToken.Value(), updateToken);
+        requestorCore->mUpdateVersion = response.softwareVersion.Value();
```

Comment:
What guarantees that `response.softwareVersion.HasValue()` here?

Reply:
Actually, nothing. I thought of adding a method to validate the whole response later, but maybe I can add it to this PR.

Reply:
Added the validation of the response to make sure all mandatory fields (for the EMBER_ZCL_OTA_QUERY_STATUS_UPDATE_AVAILABLE case) are present.